### ISSUE TITLE
SelectMany with result projection isn't supported

### ISSUE THREAD

**Issue by thomaslevesque** [Azure/azure-cosmos-dotnet-v3#683]
(already [reported in v2](https://github.com/Azure/azure-cosmos-dotnet-v2/issues/608), issue is still here in v3)

**Describe the bug**

A query that uses the [`SelectMany` method](https://docs.microsoft.com/en-us/dotnet/api/system.linq.queryable.selectmany?view=netframework-4.7.2#System_Linq_Queryable_SelectMany__3_System_Linq_IQueryable___0__System_Linq_Expressions_Expression_System_Func___0_System_Collections_Generic_IEnumerable___1____System_Linq_Expressions_Expression_System_Func___0___1___2___) with 3 arguments throws the following `DocumentQueryException`:

> Incorrect number of arguments for method 'SelectMany'. Expected '2' but received '3'., Windows/10.0.18362 cosmos-netstandard-sdk/3.1.4

There's an obvious SQL translation, which should be supported.

**To Reproduce**
Execute a LINQ query like this:

```csharp
var query =
    from order in container.GetItemLinqQueryable<Order>()
    from line in order.OrderLines
    ...
```

**Expected behavior**
The query should be translated to something like this:

```sql
SELECT ...
FROM o
JOIN line IN o.OrderLines
...
```

**Actual behavior**
Executing the query throws the following `DocumentQueryException`:

> Incorrect number of arguments for method 'SelectMany'. Expected '2' but received '3'., Windows/10.0.18362 cosmos-netstandard-sdk/3.1.4

**Environment summary**
SDK Version: Microsoft.Azure.Cosmos 3.1.1
OS Version (e.g. Windows, Linux, MacOSX): Windows 10

**Comment by Imbaker1234** [comment]
Agreed. Without the ability to query hypothetical nodes on our documents the entire point of using the LINQ provider to benefit from deferred execution seems well...pointless. Additionally, unless I'm missing something this seems like it should be easy enough to do.

**Comment by CerosDev** [comment]
So any update on that?

**Comment by lcharanteja** [comment]
Update ?

**Comment by acandylevey** [comment]
You also get this error when doing:

`.SelectMany(order => order.orderLines, (order, orderLine) => order);`

**Comment by AGuirao-Atlas** [comment]
Any new on this front? as @Imbaker1234 said, its pointless to use linq for complez scenarios without this
